lib/types/rules/ecmascript-6.d.ts

```diff
@@ -231,6 +231,52 @@ export interface ECMAScript6 extends Linter.RulesRecord {
      */
     "no-new-symbol": Linter.RuleEntry<[]>;
 
+    /**
+     * Rule to disallow specified names in exports.
+     *
+     * @since 7.0.0-alpha.0
+     * @see https://eslint.org/docs/rules/no-restricted-exports
+     */
+    "no-restricted-exports": Linter.RuleEntry<
+        [
+            Partial<{
+                /**
+                 * @default []
+                 */
+                restrictedNamedExports: string[];
+                /**
+                 * @since 9.3.0
+                 */
+                restrictedNamedExportsPattern: string;
+                /**
+                 * @since 8.33.0
+                 */
+                restrictDefaultExports: Partial<{
```

Comment:
The option works only if the r`estrictedNamedExports` option does not contain the `"default"` value. Is it possible to have that check via typescript too?

Reply:
Not really 😕. There are no "negated" types in TypeScript (https://github.com/microsoft/TypeScript/issues/4196), so you can't say _"this type is all strings **except** some literal"_.

Reply:
Hmm, it should be fine then.